packages/discovery-react-components/src/components/DocumentPreview/utils/documentData.ts

```diff
@@ -58,7 +58,7 @@ export function detectPreviewType(document: DiscoveryDocument, file?: DocumentFi
 
   const isJsonType = isJsonFile(document);
   const isCsvType = isCsvFile(document);
-  if (document.html && !isJsonType && !isCsvType) {
+  if (document.html && hasTextMappings && !isJsonType && !isCsvType) {
```

Comment:
I think that we can still render `HTML` when we don't have `passage`. That's similar to PDF.